src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -815,6 +895,46 @@ def main():
     if showtiming:
         print("... in %s seconds." % (time.time() - start_time))
 
+    # STEP 6
+    data_assimilation_csv = data_assimilation_parameters.get(
+        "data_assimilation_csv", None
+    )
+    data_assimilation_filter = data_assimilation_parameters.get(
+        "data_assimilation_filter", None
+    )
+    if data_assimilation_csv or data_assimilation_filter:
+        if showtiming:
+            start_time = time.time()
+        if verbose:
+            print("creating usgs time_slice data array ...")
+
+        usgs_df = nnu.build_data_assimilation(data_assimilation_parameters)
+
+        if verbose:
+            print("usgs array complete")
+        if showtiming:
+            print("... in %s seconds." % (time.time() - start_time))
+
+    else:
+        usgs_df = pd.DataFrame()
+
+    # if showtiming:
+    #     start_time = time.time()
+    # if verbose:
+    #     print("creating usgs time_slice data array ...")
+
+    # if data_assimilation_csv:
+    #     usgs_df = nnu.build_data_assimilation_csv(data_assimilation_parameters)
+    # elif data_assimilation_filter:
+    #     usgs_df = nnu.build_data_assimilation_folder(data_assimilation_parameters)
+    # else:
+    #     usgs_df = pd.DataFrame()
+
+    # if verbose:
+    #     print("usgs array complete")
+    # if showtiming:
+    #     print("... in %s seconds." % (time.time() - start_time))
```

Comment:
- [ ] We can delete these comments after we test everything and before merging the PR.